service/matching/handler/engine.go

```diff
@@ -1202,6 +1196,130 @@ func (e *matchingEngineImpl) emitInfoOrDebugLog(
 	}
 }
 
+func (e *matchingEngineImpl) shutDownNonOwnedTasklists() error {
+	if !e.config.EnableTasklistOwnershipGuard() {
+		return nil
+	}
+	noLongerOwned, err := e.getNonOwnedTasklistsLocked()
+	if err != nil {
+		return err
+	}
+
+	for _, tl := range noLongerOwned {
+		// for each of the tasklists that are no longer owned, kick off the
+		// process of stopping them. The stopping process is IO heavy and
+		// can take a while, so do them in parallel so as to not hold the
+		// lock too long.
```

Comment:
which lock is being held here? If this comment belongs to a previous iteration then let's just stop the tasklists synchronously. Better to not return from `shutDownNonOwnedTasklists` while some background goroutines are still stopping task lists. `shutDownNonOwnedTasklists` may be called again via ticker and might try to do the same/overlapping work and cause unexpected problems.

Reply:
It does refer to an earlier iteration where it was locking, so let me fix. 

My concern with doing it serially is that the net time to shut down will be quite slow, the `tlmgr.Close()` method does a lot of IO. What about throwing in a waitgroup and doing it pararallel?

Reply:
doing parallel should be fine I think. just avoid leaving them behind before returning